### PR TITLE
Improved: code to apply the filter on all the category pages (#561)

### LINKED ISSUE #561: Unable to apply filter on category page after changing the page
## Current behavior
When trying to apply a filter on the category page after changing the page unable to do so.

## Expected behavior
Filters should be applied on all the category pages.

## Steps to reproduce the issue
1. Go to any category
2. Change the page
3. Try to apply a filter
4. You will see that the products are not updated.

## Can you handle fixing this bug by yourself?

- [x] YES
- [ ] NO

## Environment details

- Browser: 
- OS: 
- Code Version:

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Related Issues
#561 

Closes #561 

### Short Description and Why It's Useful
The changes will remove the page number from the route which is causing the page change instead of product update.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ymaheshwari1** [comment]
Maintained the changelog for this PR here: https://github.com/vuestorefront/vsf-capybara/pull/573

File: pages/Category.vue [hunk 1]
```diff
@@ -530,6 +530,7 @@ export default {
       }
     },
     changeFilter (filter) {
+      delete this.$route.query.page
```
Comment (Fifciu) [inline pages/Category.vue:533]: I think it is not a proper approach. You shouldn't modify this.$route.query object this way. Only via `push` or `replace` method. Please think about a different approach

**Review by Fifciu — CHANGES_REQUESTED** [review]

File: pages/Category.vue (hunk 1 above)
Comment (ymaheshwari1) [reply pages/Category.vue]: @Fifciu Solved the issue by updating the filterHelpers.ts file in catalog-next module.
PR: https://github.com/vuestorefront/vue-storefront/pull/5396

**Comment by ymaheshwari1** [comment]
Closing this PR as the code improvements for the issue has been handled in the vue storefront [PR](https://github.com/vuestorefront/vue-storefront/pull/5396).

**Pull request closed without merging**